simpleclient_executor/src/main/java/io/prometheus/client/executor/InstrumentedExecutorService.java

```diff
@@ -0,0 +1,204 @@
+package io.prometheus.client.executor;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Gauge;
+import io.prometheus.client.Histogram;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class InstrumentedExecutorService implements ExecutorService {
+
+  private static final AtomicLong NAME_COUNTER = new AtomicLong();
+  private static final Counter ST_COUNTER;
+  private static final Counter CT_COUNTER;
+  private static final Gauge R_GAUGE;
+  private static final Histogram D_HISTOGRAM;
+
+  private final ExecutorService delegate;
+  private final Counter.Child submittedCounter;
+  private final Counter.Child completedCounter;
+  private final Gauge.Child runningGauge;
+  private final Histogram.Child durationHistogram;
+
+  static {
+    ST_COUNTER = Counter.build()
+        .name("instrumented_executor_service_submitted_count")
+        .labelNames("serviceName")
+        .help("Instrumented executor service submitted total count")
+        .register();
+    CT_COUNTER = Counter.build()
+        .name("instrumented_executor_service_completed_count")
+        .labelNames("serviceName")
+        .help("Instrumented executor service completed total count")
+        .register();
+
+    R_GAUGE = Gauge.build()
+        .name("instrumented_executor_service_running_count")
+        .labelNames("serviceName")
+        .help("Instrumented executor service running total count")
+        .register();
+
+    D_HISTOGRAM = Histogram.build()
```

Comment:
A histogram has buckets to cover 1ms-10s by default, is that appropriate for this sort of usage?

A quantile-less summary is probably a better choice.